src/core/js/views/buttonsView.js

```diff
@@ -1,184 +1,180 @@
-                // the inability to represent selected incorrect/correct answers to a screen reader, may need revisiting
-                switch (changedAttribute) {
-                    case BUTTON_STATE.SHOW_CORRECT_ANSWER:
-                    case BUTTON_STATE.HIDE_CORRECT_ANSWER:
-
-                        Adapt.a11y.toggleAccessible($buttonsAction, false);
-
-                }
-
-            }
-        },
-
-        checkFeedbackState: function(){
-            var canShowFeedback = this.model.get('_canShowFeedback');
-
-            this.$('.buttons-action').toggleClass('buttons-action-fullwidth buttons-action-enlarge', !canShowFeedback);
-            this.$('.buttons-feedback').toggleClass('no-feedback', !canShowFeedback);
-            this.$('.buttons-marking-icon').toggleClass('no-feedback', !canShowFeedback);
-        },
-
-        updateAttemptsCount: function(model, changedAttribute) {
-            var isInteractionComplete = this.model.get('_isInteractionComplete');
-            var attemptsLeft = (this.model.get('_attemptsLeft')) ? this.model.get('_attemptsLeft') : this.model.get('_attempts');
-            var isCorrect = this.model.get('_isCorrect');
-            var shouldDisplayAttempts = this.model.get('_shouldDisplayAttempts');
-            var attemptsString;
-
-            this.checkResetSubmittedState();
+  //convert BUTTON_STATE to property name
+  var textPropertyName = {
+    "SUBMIT": "submit",
+    "CORRECT": "correct",
+    "INCORRECT": "incorrect",
+    "SHOW_CORRECT_ANSWER": "showCorrectAnswer",
+    "HIDE_CORRECT_ANSWER": "hideCorrectAnswer",
+    "SHOW_FEEDBACK": "showFeedback",
+    "RESET": "reset"
+  };
+
+  var ButtonsView = Backbone.View.extend({
+
+    initialize: function(options) {
+      this.parent = options.parent;
+
+      this.listenTo(Adapt, 'remove', this.remove);
+      this.listenTo(this.model, 'change:_buttonState', this.onButtonStateChanged);
+      this.listenTo(this.model, 'change:feedbackMessage', this.onFeedbackMessageChanged);
+      this.listenTo(this.model, 'change:_attemptsLeft', this.onAttemptsChanged);
+      this.render();
+    },
+
+    events: {
+      'click .js-btn-action': 'onActionClicked',
+      'click .js-btn-feedback': 'onFeedbackClicked'
+    },
+
+    render: function() {
+      var data = this.model.toJSON();
+      var template = Handlebars.templates['buttons'];
+      _.defer(_.bind(function() {
+        this.postRender();
+        Adapt.trigger('buttonsView:postRender', this);
+      }, this));
+      this.$el.html(template(data));
+    },
+
+    postRender: function() {
+      this.refresh();
+    },
+
+    checkResetSubmittedState: function() {
+      var isSubmitted = this.model.get('_isSubmitted');
+
+      if (!isSubmitted) {
+
+        var $icon = this.$('.js-btn-marking');
+        $icon.removeClass('is-incorrect');
+        $icon.removeClass('is-correct');
+        $icon.addClass('u-display-none');
```

Comment:
could just do this in one line i.e.
```suggestion
        this.$('.js-btn-marking').removeClass('is-incorrect is-correct').addClass('u-display-none');
```

Reply:
Done